che-theia-factory/src/browser/factory-theia-client.ts

```diff
@@ -52,31 +106,29 @@ export class FactoryTheiaClient implements FrontendApplicationContribution {
             return;
         }
 
-        const cheApiExternalVar = this.getEnvVariable('CHE_API_EXTERNAL');
-        if (!cheApiExternalVar) {
-            return;
-        }
-
         const projectsRootEnvVar = this.getEnvVariable('CHE_PROJECTS_ROOT');
-        var projectsRoot = "/projects";
         if (projectsRootEnvVar && projectsRootEnvVar.value) {
-            projectsRoot = projectsRootEnvVar.value;
+            this.projectsRoot = projectsRootEnvVar.value;
         }
 
-        const factory = new FactoryService(FactoryTheiaClient.axiosInstance, String(cheApiExternalVar.value));
-        const response: AxiosResponse<IFactory> = await factory.getById<IFactory>(factoryid);
+        this.onAppClosedActions = this.factoryManager.getFactoryOnAppClosedActions(factory);
+        this.onAppLoadedActions = this.factoryManager.getFactoryOnAppLoadedActions(factory);
 
-        response.data.workspace.projects.forEach((project: IProjectConfig) => {
+        const onProjectsLoadedActions = this.factoryManager.getFactoryOnProjectsLoadedActions(factory);
+        const importProjectPromices: Array<Promise<void>> = [];
+
+        const projects = this.factoryManager.getFactoryProjects(factory);
+        projects.forEach((project: IProjectConfig) => {
             if (!project.source) {
                 return;
             }
 
             const source = project.source;
-            const projectPath = projectsRoot + project.path;
+            const projectPath = this.projectsRoot + project.path;
 
-            this.messageService.info(`Cloning ... ${source.location} to ${projectPath}...`)
+            this.messageService.info(`Cloning ... ${source.location} to ${projectPath}...`);
 
-            this.git.clone(
+            importProjectPromices.push(this.git.clone(
```

Comment:
We could use "await" here. But up to you.

Reply:
I am sure about that. We should start in parallel and wait for resolve importProjectPromices.